source/MaterialXCore/Node.cpp

```diff
@@ -656,13 +656,33 @@ InputPtr Node::addInputFromNodeDef(const string& name)
         InputPtr nodeDefInput = nodeDef ? nodeDef->getActiveInput(name) : nullptr;
         if (nodeDefInput)
         {
-            nodeInput = addInput(nodeDefInput->getName());
-            nodeInput->copyContentFrom(nodeDefInput);
+            nodeInput = addInput(nodeDefInput->getName(), nodeDefInput->getType());
+            nodeInput->setAttribute(ValueElement::VALUE_ATTRIBUTE, 
+                                    nodeDefInput->getAttribute(ValueElement::VALUE_ATTRIBUTE));
         }
     }
     return nodeInput;
 }
 
+void Node::addInputsFromNodeDef()
+{
+    NodeDefPtr nodeNodeDef = getNodeDef();
+    if (nodeNodeDef)
+    {
+        for (auto nodeDefInput : nodeNodeDef->getActiveValueElements())
+        {
+            const string& name = nodeDefInput->getName();
+            InputPtr nodeInput = getInput(name);
+            if (!nodeInput)
+            {
+                nodeInput = addInput(name, nodeDefInput->getType());
```

Comment:
Believe this is as compact as I can get this to be now :).
All the extra parameters are avoided now.

Reply:
This looks great, thanks Bernard!